app/src/main/java/com/android/unio/model/notification/NotificationScheduler.kt

```diff
@@ -0,0 +1,240 @@
+import androidx.core.os.toPersistableBundle
+import androidx.work.Data
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.android.unio.MainActivity
+import com.android.unio.model.strings.NotificationStrings.NOTIFICATION_SCHEDULER_TYPE_CANCEL
+import com.android.unio.model.strings.NotificationStrings.NOTIFICATION_SCHEDULER_TYPE_CREATE
+
+class NotificationScheduler() : BroadcastReceiver() {
+
+  companion object {
+    /**
+     * Creates a PendingIntent for the notification.
+     *
+     * @param context The application context.
+     * @param data The data containing notification details.
+     * @return The created PendingIntent.
+     */
+    private fun createIntent(context: Context, data: Data): PendingIntent {
+
+      val notificationId = data.getInt("notificationId", 0)
+      val bundle = Bundle(data.keyValueMap.toPersistableBundle())
+      val alarmIntent =
+          Intent(context.applicationContext, NotificationScheduler::class.java).putExtras(bundle)
+      // pass intent if extra data is needed
+      val pendingIntent =
+          PendingIntent.getBroadcast(
+              context,
+              notificationId,
+              alarmIntent,
+              PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
+
+      return pendingIntent
+    }
+    /**
+     * Schedules a notification using AlarmManager.
+     *
+     * @param context The application context.
+     * @param data The data containing notification details including timeMillis.
+     */
+    fun schedule(context: Context, data: Data) {
+      val manager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+      // pass intent if extra data is needed
+      val pendingIntent = createIntent(context, data)
+      val timeMillis = data.getLong("timeMillis", System.currentTimeMillis() + 10000)
+      println(timeMillis)
+      manager.set(AlarmManager.RTC_WAKEUP, timeMillis, pendingIntent)
+      Log.d("NotificationScheduler", "Notification scheduled")
+    }
+    /**
+     * Unschedules a previously scheduled notification using AlarmManager.
+     *
+     * @param context The application context.
+     * @param data The data containing notification details.
+     */
+    fun unschedule(context: Context, data: Data) {
+      val manager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+
+      val pendingIntent = createIntent(context, data)
+      manager.cancel(pendingIntent)
+      Log.d("NotificationScheduler", "Notification canceled")
+    }
+  }
+  /**
+   * Receives the broadcast and triggers the notification or handles BOOT_COMPLETED action.
+   *
+   * @param context The application context.
+   * @param intent The received broadcast intent.
+   */
+  override fun onReceive(context: Context?, intent: Intent?) {
+    if (intent == null) {
+      Log.w("NotificationScheduler", "Received a null Intent")
```

Comment:
This would happen if a broadcast is triggered with a null intent, which i guess is possible because the onReceive method is defined like this by the BroadcastReceiver java class. I don't know in which case it would happen so I guess we could log it as an error.. We can talk about it tomorrow if you want.